docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,130 @@
+# Faker connector
+
+The Faker connector generates realistically looking random data matching a
+specific schema.
+
+## Configuration
+
+To configure the Faker connector, create a catalog properties file
+`etc/catalog/example.properties` with the following contents:
+
+```text
+connector.name=faker
+```
+
+## Examples
+
+Faker generates data when reading from a table created in a catalog using this
+connector. This makes it easy to fill an existing schema with random data, by
+copying only the schema into a Faker catalog, and inserting the data back into
+the original tables.
+
+```sql
+CREATE TABLE faker.default.customer (LIKE production.public.customer EXCLUDING PROPERTIES);
+
+INSERT INTO production.public.customers
+SELECT *
+FROM faker.default.customers
+WHERE
+  born_at BETWEEN CURRENT_DATE - INTERVAL '150' YEAR AND CURRENT_DATE
+  AND age_years BETWEEN 0 AND 150
+LIMIT 100;
+```
+
+To generate more realistic data, choose specific generators by setting the
+`generator` property on columns:
+
+```sql
+SHOW CREATE TABLE production.public.customers;
+
+-- copy the output of the above query and add some properties:
+CREATE TABLE faker.default.customer (
+  id UUID NOT NULL,
+  name VARCHAR NOT NULL WITH (generator = '#{Name.first_name} #{Name.last_name}'),
+  address VARCHAR NOT NULL WITH (generator = '#{Address.fullAddress}'),
+  born_at DATE,
+  age_years INTEGER
+);
+```
+
+See the Datafaker's documentation for more information about
+[the expression](https://www.datafaker.net/documentation/expressions/) syntax
+and [available providers](https://www.datafaker.net/documentation/providers/).
+
+## Generators
+
+A particular data generator is selected based on the column type.
+
+For `CHAR`, `VARCHAR`, and `VARBINARY` column, the default generator uses the
+`Lorem ipsum` placeholder text. Unbounded columns return a random sentence with
```

Comment:
Link to that on wikipedia or whereever.

Reply:
Maybe we should have a table of all supported data types and what is happening and supported in terms of defining bounds, default values, invalid values, enums and whatever

Reply:
I listed the unsupported data types in the `Type mapping` section, since there are very few of them, and supporting them is just not practical. All the supported data types are grouped into character and non-character types, and share the same properties, that's why I haven't added a table or list.